src/main/org/pantsbuild/jarjar/util/JarTransformer.java

```diff
@@ -20,17 +20,35 @@
 import org.objectweb.asm.ClassReader;
 import org.objectweb.asm.ClassVisitor;
 import org.objectweb.asm.ClassWriter;
+import org.pantsbuild.jarjar.misplaced.MisplacedClassStrategy;
+
+abstract public class JarTransformer implements JarProcessor {
+
+    private MisplacedClassStrategy misplacedClassStrategy;
+
+    public JarTransformer(MisplacedClassStrategy misplacedClassStrategy) {
+        this.misplacedClassStrategy = misplacedClassStrategy;
+    }
 
-abstract public class JarTransformer implements JarProcessor
-{
     public boolean process(EntryStruct struct) throws IOException {
         if (struct.name.endsWith(".class")) {
+            String originalClassName = null;
             ClassReader reader;
             try {
                 reader = new ClassReader(struct.data);
+                originalClassName = reader.getClassName() + ".class";
             } catch (Exception e) {
+                e.printStackTrace();
                 return true; // TODO?
```

Comment:
Yeah this is what was previously just eating the exceptions when jarjar failed to shade java8 classes.